old/paper-input.html

```diff
@@ -0,0 +1,183 @@
+<!--
+    @license
+    Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+    This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+    The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+    The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+    Code distributed by Google as part of the polymer project is also
+    subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+
+<!--
+@element paper-input
+-->
+
+<link rel="import" href="paper-input-underline.html">
+
+<dom-module id="paper-input">
+
+  <style>
+
+    :host {
+      display: block;
+    }
+
+    .content ::content input,
+    .content ::content textarea {
+      margin: 0;
+      padding: 0.5em 0 0.25em;
+      border: none;
+      font: inherit;
+      width: 100%;
+    }
+
+    ::-webkit-input-placeholder {
+      color: #757575;
+    }
+
+    ::-moz-placeholder {
+      color: #757575;
+      opacity: 1;
+    }
+
+    :-ms-input-placeholder {
+      color: #757575;
+    }
+
+    paper-input-underline {
+      margin-bottom: 0.25em;
+    }
+
+    .add-on {
+      color: #757575;
+    }
+
+    :host([focused]) .add-on {
+      color: #4059a9;
+    }
+
+    :host([invalid][focused]) .add-on {
+      color: #d34336;
+    }
+
+  </style>
+
+  <template>
+
+    <div class="content">
+      <content select="input,textarea"></content>
+    </div>
+
+    <paper-input-underline id="underline"></paper-input-underline>
+
+    <content select=".add-on"></content>
+
+  </template>
+
+</dom-module>
+
+<script>
+
+(function() {
+
+  Polymer({
+
+    is: 'paper-input',
```

Comment:
Does this element need to do more ARIA work? It kind of tries to reflect focus state. Should it also take over the ARIA semantics from the `<content>` that it is composing? :surfer: